src/routes/spotify/desktop/Attract.svelte

```diff
@@ -27,6 +29,29 @@
       message = messages[(i + 1) % messages.length];
     }, 5000);
   }
+
+  const shuffleArray = <T,>(array: T[]): T[] =>
+    array
+      .map((value) => ({ value, sort: Math.random() }))
+      .sort((a, b) => a.sort - b.sort)
+      .map(({ value }) => value);
```

Comment:
there's also a shuffle hiding inside playlist.svelte.ts. Let's make one method that both use? 

`export const shuffleArray = <T>(array: T[]): T[] => ...` in `lib/array.ts` perhaps

Reply:
There's also a `chunk` func in there that could be extracted

Reply:
oooo, hell yeah extracted both shuffle and chunk 👌 

had to rename the import in `playlist.svelte` to avoid conflicting with `shuffle` in state, let me know if ideas for a better name there!